src/main/scala/nl.knaw.dans.easy.solr4files/components/DDM.scala

```diff
@@ -85,6 +96,22 @@ object DDM {
     "https://easy.dans.knaw.nl/schemas/vocab/2015/narcis-type.xsd"
   )("Discipline")
 
+  def spacedText(n: Node): String = {
+    val s = mutable.ListBuffer[String]()
+
+    def strings(n: Seq[Node]): Unit =
```

Comment:
this looks and feels like recursion. You can easily turn this into tail recursion. Hint: pattern matching on `n`.

Reply:
I was thinking about this some more, if you don't mind...

```scala
def spacedText(ns: Seq[Node]): String = {

  @tailrec
  def internal(ns: Seq[Node], result: List[String] = List.empty): List[String] = {
    ns match {
      case Seq() => result
      case Seq(h, t@_*) if h.child.isEmpty => internal(t, h.text :: result)
      case Seq(h, t@_*) => internal(h.child ++ t, result)
    }
  }

  internal(ns).reverse.mkString(" ")
}
```

Reply:
brief explanation:
* I add the `h.text` to the front of the list (prepend), since `::` is an O(1) operation. This in contrast to adding to the end of the list (append), which I'm not sure about, but I believe it's like O(n).
* To make sure the desired order is correct, I have to do a `reverse` at the end. Note again that it is way more efficient to do a single `reverse` operation at the end than do an append for every node in the tree!